testing/demo/pkg/transfer/update_groth16_client.go

```diff
@@ -71,37 +73,78 @@ func updateGroth16LightClient(evmTransferBlockNumber uint64) error {
 	return nil
 }
 
+// DecodePublicValues decodes the public values from the Blevm proof aggregator.
+// The public values are encoded using bincode serialization. They're encoded in fixed order.
+// All of the fields are fixed size bytes32 and are encoded in little endian format.
+// Celestia header hashes is not a fixed size array therefore it is encoded
+// as a u64 length prefix(bytes8) followed by the hashes.
```

Comment:
I have a question wrt to this encoding. In the prover-service I can see:
```rust
let response = ProveStateTransitionResponse {
    proof: aggregation_output.proof.bytes(),
    public_values: aggregation_output.proof.public_values.to_vec(),
};
```

Is bincode serialization inferred implicitly somewhere for public_values?